engine/table/src/main/java/io/deephaven/engine/table/impl/locations/TableDataService.java

```diff
@@ -20,6 +20,22 @@ public interface TableDataService {
     @NotNull
     TableLocationProvider getTableLocationProvider(@NotNull TableKey tableKey);
 
+    /**
+     * Request a raw {@link TableLocationProvider} from this service. A raw {@link TableLocationProvider} does not
+     * compose and/or delegate to other {@link TableLocationProvider}s.
+     *
+     * @param tableKey The {@link TableKey} to lookup
+     * @param tableLocationKey The {@link TableLocationKey} to lookup
+     * @return A raw {@link TableLocationProvider} for the specified {@link TableKey} and {@link TableLocationKey}, or
+     *         null if either key is not present
```

Comment:
I think this needs more definition (and the clarification can be in the summary above).
Is this method getting "the TLP that currently has the location for the TK/TLK", or "the TLP that would provide the location for TK/TLK"? The implementations below are inconsistent, and the answer impacts the default impl.

Reply:
"Provide", not "has". I've adopted Ryan's doc suggestion that I think improves this.